### PR TITLE
Add  `meta:enum` descriptions for task types

### LINKED ISSUE #245: document and revisit `taskType`
_Originally posted by @jkowalleck in https://github.com/CycloneDX/specification/pull/222#discussion_r1211618704_
            

see also: https://github.com/CycloneDX/specification/pull/222#issuecomment-1570077893

@mrutkows reminded: should reflect CI/CD stages. There might be some standardization work done for the wordings.

this ticket is about adding documentation for TaskType and its Enum values.
see https://github.com/CycloneDX/specification/blob/8af880d5f2ba0a107de88a920a76cedd5ba75083/schema/bom-1.5.schema.json#L3753

### PR DESCRIPTION
**Pull request by mrutkows**
Fixes #245

- [x] update JSON docs
- [x] update XML docs
- [x] update ProtoBuff docs

### CODE REVIEW

**Comment by mrutkows** [comment]
@jkowalleck Please verify I did this correctly and to your satisfaction. Thanks!

**Comment by jkowalleck** [comment]
@mrutkows looks promissing.

I've already marked this PR for milestone 1.6 and linked the related issue.
Please add the "documentation" label to the PR, so reviewers can spot it easily.
You will find this in GitHub's right panel.

I marked is as "draft", since I found some tasks missing.
Please add/check also the XML docs and ProtoBuff docs. if all is done, check the boxed and tasks I've just added to the PR description. If all is done, click the "ready for review" button, so the "draft" mark gets removed, and everybody is aware that this is ready for review.

**Comment by mrutkows** [comment]
@jkowalleck Thanks Jan, was not aware that I needed to change XSD and Protobuf as well; will work on now. However, AFAIK I do have descriptions for all task types in the current enum. (just double checked).  Which one(s) are missing from the JSON schema committed?

File: schema/bom-1.6.proto [hunk 1]
```diff
@@ -1482,17 +1482,29 @@ message Condition {
 
 enum TaskType {
   TASK_TYPE_COPY = 0;
+  // A task that copies software or data used to accomplish other tasks in the workflow.
```
Comment (jkowalleck) [inline schema/bom-1.6.proto:1485]: Comments go over the accosted field,
not under them.
please change

**Review by jkowalleck — CHANGES_REQUESTED** [review]

File: schema/bom-1.6.proto (hunk 1 above)
Comment (mrutkows) [reply schema/bom-1.6.proto]: Sorry for that... 3rd time is a charm ;)

**Review by stevespringett — APPROVED** [review]
LGTM. Thanks @mrutkows. @jkowalleck, your thought?

File: schema/bom-1.6.proto (hunk 1 above)
Comment (jkowalleck) [reply schema/bom-1.6.proto:1484]: LGTM

**Review by jkowalleck — APPROVED** [review]

**Comment by jkowalleck** [comment]
> LGTM. Thanks @mrutkows. @jkowalleck, your thought?

all fine now. I will merge it, then.

**Pull request merged**